### PR TITLE
Bug 651

### PR DESCRIPTION
**Pull request by jroy1992**
Separating validations for comp on strict checking of frame range.

### CODE REVIEW

File: hooks/tk-multi-publish2/nuke/publish_files_comp.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+    def _framerange_to_be_published(self, item):
+        """
+        Since users have the option to render only a subset of frames,
+        adding validation to check if the full frame range is being published.
+
+        :param item: Item to process
+        :return: True if yes false otherwise
+        """
+        lss_path = item.properties['node']['cached_path'].value()
```
Comment (mbrainerd) [inline hooks/tk-multi-publish2/nuke/publish_files_comp.py]: You already have this cached in item.properties["sequence_paths"] so no need to calculate again. The code should be:

```
if item.properties["is_sequence"]:
    files = item.properties["sequence_paths"]
else:
    files = item.properties["path"]
```

**Comment by jroy1992** [comment]
Fixed requested change.

File: hooks/tk-multi-publish2/nuke/publish_files_comp.py (hunk 1 above)
Comment (jroy1992) [reply hooks/tk-multi-publish2/nuke/publish_files_comp.py]: Noticed that if sequence, then also we have the path stored in item.properties["path"].
so keeping lss_path = item.properties['path']

**Comment by aparna-d2** [comment]
Merged because [sgtk_config](http://dd-git.d2.com/india/sgtk_config/merge_requests/19) changes were already merged. Discrepancies will cause errors if someone releases from sgtk_config.

File: hooks/tk-multi-publish2/nuke/publish_files.py [hunk 2]
```diff
@@ -96,7 +96,7 @@ def _sync_frame_range(self, item):
             # compare if the frame range set at root level is same as the shotgun cut_in, cut_out
             root = nuke.Root()
             if root.firstFrame() != data[in_field] or root.lastFrame() != data[out_field]:
-                self.logger.error("Frame range not synced with Shotgun.")
+                self.logger.warning("Frame range not synced with Shotgun.")
                 return False
```
Comment (aparna-d2) [inline hooks/tk-multi-publish2/nuke/publish_files.py]: You probably want to return True here to unblock Publish. But that would probably show validations passed, in which case the user won't bother to read the warning. Perhaps a pop-up message?

**Pull request merged**